### PR TITLE
Extract reusable action for test artifact upload

### LINKED ISSUE #9135: GHA: Bundle post-test steps into a reusable action
With #9129, #9035, #9130 and #9132 implemented, we will have a lot of steps that run after the tests are finished to collect reports and statistics. Since we want to keep our workflows maintainable, this would be a great opportunity to build some re-usable composite actions.

See also: https://dev.to/jameswallis/using-github-composite-actions-to-make-your-workflows-smaller-and-more-reusable-476l

### PR DESCRIPTION
**Pull request by menski**
## Description
Create composite github action to bundle test artifact upload steps.

At the moment the steps are:
- clean system-out from test XML output
- upload test artifact for later use

## Related issues

closes #9135

### CODE REVIEW

File: .github/actions/collect-test-artifacts/action.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+name: Collect test artifacts
+
+description: Collects test outputs and uploads them as artifact
+
+inputs:
+  name:
+    description: 'Name of the test artifact.'
+    required: true
+  sed-bin:
```
Comment (oleschoenburg) [inline .github/actions/collect-test-artifacts/action.yml:10]: :x: This appears to be left-over?

**Review by lenaschoenburg — APPROVED** [review]
Sorry for the delay :see_no_evil: 

LGTM, just one minor comment to remove an unused input

File: .github/actions/collect-test-artifacts/action.yml (hunk 1 above)
Comment (menski) [reply .github/actions/collect-test-artifacts/action.yml]: Sorry I missed to delete that, I tried different approaches and you are right this is a left over

**Comment by oleschoenburg** [comment]
@menski 

sorry, missed during review: we need an `if: always()` wherever we use the new `collect-test-artifacts` action, otherwise we don't upload on test failure.
Previously, this worked because the last step had an `if: always()`, now it doesn't because the entire composite action is skipped on failure.

**Pull request merged**